stan/math/prim/arr/fun/append_array.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef STAN_MATH_PRIM_ARR_FUN_APPEND_ARRAY_HPP
+#define STAN_MATH_PRIM_ARR_FUN_APPEND_ARRAY_HPP
+
+#include <vector>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Concatenate two vectors into one.
+     *
+     * @tparam T Type of elements contained in vector.
+     * @param x First vector.
+     * @param y Second vector.
+     * @return A vector of x and y concatenated together (in that order).
+     */
+    template <typename T>
+    inline std::vector<T> append_array(std::vector<T> x, std::vector<T> y) {
```

Comment:
The arguments need to be declared as `const std::vector<T>&` --- a reference to a constant.